kuma/scrape/fixture.py

```diff
@@ -0,0 +1,327 @@
+        },
+        'search.filter': {
+            'natural_key': ('name', 'slug'),
+            'relations': {
+                'group': {
+                    'link': 'to_one',
+                    'resource': 'search.filtergroup'
+                },
+                'tags': {
+                    'link': 'to_many',
+                    'resource': 'taggit.tag'
+                },
+            },
+        },
+        'search.filtergroup': {
+            'natural_key': ('name', 'slug'),
+        },
+        'sites.site': {
+            'natural_key': ('id',),
+        },
+        'socialaccount.socialaccount': {
+            'natural_key': ('uid', 'provider'),
+            'relations': {
+                'user': {
+                    'link': 'to_one',
+                    'resource': 'users.user',
+                }
+            }
+        },
+        'socialaccount.socialapp': {
+            'natural_key': ('name', ),
+            'relations': {
+                'sites': {
+                    'link': 'to_many',
+                    'resource': 'sites.site'
+                }
+            }
+        },
+        'taggit.tag': {
+            'natural_key': ('name',),
+        },
+        'users.user': {
+            'natural_key': ('username',),
+            'relations': {
+                'groups': {
+                    'link': 'to_many',
+                    'resource': 'auth.group',
+                },
+            },
+            'filters': {
+                'password': 'make_password',
+            },
+        },
+        'users.userban': {
+            'natural_key': ('user', 'by'),
+            'relations': {
+                'user': {
+                    'link': 'to_one',
+                    'resource': 'users.user',
+                },
+                'by': {
+                    'link': 'to_one',
+                    'resource': 'users.user',
+                },
+            },
+        },
+        'waffle.flag': {
+            'natural_key': ('name',),
+        },
+        'waffle.switch': {
+            'natural_key': ('name',),
+        },
+    }
+
+    class NeedsDependency(Exception):
+        """A fixture has an un-resolved dependency."""
+        pass
+
+    def __init__(self, specification):
+        """Intialize with a specification dictionary."""
+        self.instances = {}
+        self.spec = self.parse_specification(specification)
+
+    def parse_specification(self, specification):
+        """Parse and validate the specification."""
+        parsed = {}
+        for model_id, items in specification.items():
+            # Parse and validate the model
+            metadata = self.model_metadata[model_id]
+            natural_key_spec = metadata['natural_key']
+            relations = metadata.get('relations', {})
+            filters = metadata.get('filters', {})
+            app_label, model_name = model_id.split('.')
+            assert apps.get_model(app_label, model_name)
```

Comment:
Thanks, this worked. I didn't know that ``apps.get_model`` was flexible that way.